src/Symfony/Bundle/SecurityBundle/Security/FirewallConfig.php

```diff
@@ -28,7 +28,20 @@ class FirewallConfig
     private $userChecker;
     private $listeners;
 
-    public function __construct($name, $requestMatcher, $securityEnabled = true, $stateless = false, $provider = null, $context = null, $entryPoint = null, $accessDeniedHandler = null, $accessDeniedUrl = null, $userChecker = null, $listeners = array())
+    /**
+     * @param string      $name
+     * @param string      $requestMatcher
+     * @param bool        $securityEnabled
+     * @param bool        $stateless
+     * @param string|null $provider
+     * @param string|null $context
+     * @param string|null $entryPoint
+     * @param string|null $accessDeniedHandler
+     * @param string|null $accessDeniedUrl
+     * @param string      $userChecker
```

Comment:
Isn't it a bit strange to have this mandatory argument after all those optional nullable arguments?